bob-apps/bin/bobp.rs

```diff
@@ -579,9 +651,69 @@ async fn put_worker(net_conf: NetConfig, task_conf: TaskConfig, stat: Arc<Statis
     }
 }
 
+fn prepare_exist_keys(from: u64, count: u64, packet_size: u64, shuffle: bool) -> Vec<Vec<u64>> {
+    let mut keys = Vec::new();
+    let mut current_low = from;
+    for _ in 0..count {
+        let mut keys_portion: Vec<_> = 
+            (current_low..current_low + packet_size).collect();
+        if shuffle {
+            keys_portion.shuffle(&mut thread_rng());
+        }
+
+        keys.push(keys_portion);
+
+        current_low += packet_size;
+    }
+    keys
+}
+
+async fn exist_worker(net_conf: NetConfig, task_conf: TaskConfig, stat: Arc<Statistics>) {
+    let mut client = net_conf.build_client().await;
+
+    let options = task_conf.find_get_options();
+    let send_size_bytes = task_conf.packet_size * (task_conf.key_size as u64);
+    let request_count = task_conf.count / task_conf.packet_size;
```

Comment:
This operation will round the number of keys up to a multiple of the package size, which is incorrect